subsys/bluetooth/audio/tbs.c

```diff
@@ -628,6 +632,73 @@ static void net_buf_put_current_calls(const struct tbs_inst *inst, struct net_bu
 	}
 }
 
+static bool is_tbs_uri_unique(const char *uri_list, const char *uri_to_search)
+{
+	bool is_matched = false;
+	size_t uri_len = strlen(uri_to_search);
+	char *uri_pos = strstr(uri_list, uri_to_search);
+
+	/* Below check is to avoid match for string like "aaa" with "aaas" or "saaa" string present
+	 * in uri_list.
+	 */
+
+	if ((uri_pos != NULL) && ((uri_pos == uri_list) || (!isalnum(*(uri_pos - 1)))) &&
+	    (!isalnum(*(uri_pos + uri_len)))) {
+		is_matched = true;
+	}
+
+	return is_matched;
```

Comment:
Did you consider using `strtok` here as well, instead of the somewhat hard to read `if` statement? 

```suggestion
	/* Split URI list by comma */
	const char *uri = uri_list(str, ",");
	
	/* Check each URI in URI list if they are identical to uri_to_search */
	while (uri != NULL) {
		if (strcmp(uri, uri_to_search) == 0) {
			return false;
		}
	
		uri = strtok(NULL, ",");
	}
   
	return true;
```

Basically applying the same approach for each URI in `uri_list` as you do for `svc_insts[i].uri_scheme_list`

Reply:
I think I understand your concern, and that is valid. 

Doing operations like `net_buf_simple_add_mem(buf, inst->uri_scheme_list, strlen(inst->uri_scheme_list));` won't add NULL termination of the buffer. 

Adding the NULL terminator is a fairly simple matter, but there isn't a `net_buf_simple_add_str` to help correctly append strings to the list of URIs. 

In many ways, I think it would be simpler (although probably more memory intensive?) to use array of strings for storage, rather than storing them comma-separated. 

So I guess that's your point with using `isalnum`, right? That it helps determine the end of the non-NULL terminated "string"? 

Perhaps using `strcat` over `net_buf_simple_add_mem` and the manually increasing the buffer length could work?